protocols/v2/binary-sv2/binary-sv2/src/lib.rs

```diff
@@ -37,6 +42,8 @@ mod test {
 
     mod test_struct {
         use super::*;
+        #[cfg(not(feature = "with_serde"))]
```

Comment:
why are these `with_serde` flags being added?

Reply:
Enforcing #![no_std] at the top of lib.rs remove the implicitly use of std::vec::Vec everywhere in the crate. So here in one of the test that is run only with feature "with_serde" the Vec must be explicitly used from alloc instead of std. Because the test is depends on with_serde feature, I made the "use alloc::vec::Vec" also depending on with_serde feature to avoid a warning if we run the test without it.

Reply:
done